File: MinerApplication/MinerApplication/GoAndEat.cpp
#include "stdafx.h"
#include "GoAndEat.h"
#include "MiningForGold.h"
GoAndEat::GoAndEat()
{
}

GoAndEat::~GoAndEat()
{
}

void GoAndEat::Execute(Miner* miner)
{
	cout << "Eating my Pasty" << endl;
	miner->m_Hunger--;

}

File: Miner Application(Unity)/Assets/Scripts/Miner/GoAndEat.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
[System.Serializable]
public class GoAndEat : State<Miner>
{
    public GoAndEat()
    {

    }
    public override void Execute(Miner agent)
    {
        agent.TargetLocation = new Vector3(2, 1, -2);
        Debug.Log("Eating my Pasty");
        agent.m_Hunger--;
    }
}
